overlord/snapstate/handlers.go

```diff
@@ -362,6 +361,12 @@ func (m *SnapManager) undoPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func installInfoUnlocked(st *state.State, snapsup *SnapSetup) (*snap.Info, error) {
+	st.Lock()
```

Comment:
is the same meaning  of Unlocked in NewTaskProgressAdapterUnlocked , I followed that.

I used the small helper because on panics (which are not unlikely in tests given how some helper are written) wrapping the call tends to produce a bit less readable stacktraces that first panic about the lock state